src/libstd/panicking.rs

```diff
@@ -41,12 +65,10 @@ use realstd::io::set_panic;
 // hook up these functions, but it is not this day!
 #[allow(improper_ctypes)]
 extern "C" {
-    fn __rust_maybe_catch_panic(
-        f: fn(*mut u8),
-        data: *mut u8,
-        data_ptr: *mut usize,
-        vtable_ptr: *mut usize,
-    ) -> u32;
+    /// The payload ptr here is actually the same as the payload ptr for the try
+    /// intrinsic (i.e., is really `*mut [u64; 2]` or `*mut *mut u8`).
+    #[unwind(allowed)]
+    fn __rust_cleanup(payload: *mut u8) -> Box<dyn Any + Send + 'static>;
```

Comment:
I can rename the symbol to that, yes. (I think we're generally of the policy that `__rust*` symbols are our own and no one should reuse that "namespace", though).

Passing the panic in by-value means that we need to duplicate the cfg-if one more time into libpanic_abort so that we can take either `[u64; 2]` or` *mut u8` which this avoids.